src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java

```diff
@@ -1,27 +1,39 @@
 package com.internet.shop.web.filter;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
 import javax.servlet.Filter;
 import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 import javax.servlet.annotation.WebFilter;
+import javax.servlet.annotation.WebInitParam;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebFilter("/*")
+@WebFilter(
+        urlPatterns = "/*",
+        initParams = @WebInitParam(name = "excluded", value = "/login,/register"))
 public class AuthenticationFilter implements Filter {
     private static final String USER_ID = "userId";
+    private static List<String> excludedServlets;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        String excluded = filterConfig.getInitParameter("excluded");
+        excludedServlets = Arrays.asList(excluded.split(","));
```

Comment:
Looks interesting. What is the difference between adding URLs manually here as we have in video and your approach (retrieve data from init parameters?).

Reply:
originally it was just an attempt to implement this mechanism in an alternative way)

but now I think that the current approach supports Single Responsibility Principle:

now we will have to change AuthenticationFilter only when the logic will be different, and not because we need to change excluded servlets

Reply:
and as we discussed, in general it is just more convenient to edit excluded servlets in web.xml